service/udp.go

```diff
@@ -420,7 +430,7 @@ func (m *natmap) Close() error {
 var maxAddrLen int = len(socks.ParseAddr("[2001:db8::1]:12345"))
 
 // copy from target to client until read timeout
-func timedCopy(clientAddr net.Addr, clientConn net.PacketConn, targetConn *natconn,
+func timedCopy(clientAddr *net.UDPAddr, proxyIP net.IP, clientConn net.PacketConn, targetConn *natconn,
```

Comment:
This could take the bound `clientConn` and the `targetConn` only, following my suggestion.

Reply:
I've changed this to take a `shadowsocksUDPWriter`.